crates/cli/src/file_utils.rs

```diff
@@ -308,16 +313,65 @@ pub fn filter_files_by_diff_aware_info(
         .collect();
 }
 
+/// Generate a fingerprint for a violation that will uniquely identify the violation. The fingerprint is calculated
+/// as is
+///  SHA2(<file-location-in-repository> - <characters-in-directory> - <content-of-code-line> - <number of characters in line>)
+pub fn get_fingerprint_for_violation(
+    violation: &Violation,
+    repository_root: &str,
+    file: &str,
+    use_debug: bool,
+) -> Option<String> {
+    let path = Path::new(repository_root).join(file);
+    if !path.exists() || !path.is_file() {
+        return None;
+    }
+    let line = violation.start.line as usize;
+
+    match read_to_string(&path) {
+        Ok(file_content) => match file_content.lines().nth(line - 1) {
+            Some(line_content) => {
+                let line_content_stripped = line_content.replace(' ', "");
+                let hash_content = format!(
+                    "{}-{}-{}-{}",
+                    file,
+                    file.len(),
+                    line_content_stripped,
+                    line_content_stripped.len()
```

Comment:
Why do we need the number of characters here and for the line above?

Reply:
Just add more entropy for the hash.

Reply:
So let's say you have the *exact* same line of code in two different files. In _some_ way, the filename turns out to have the _same_ hash. This length would just add another piece of info to try to distinguish between both.

I know it's borderline adding a tinfoil hat but that the reasoning behind it (e.g. add more stuff that could add more entropy). Happy to drop it if necessary.

Reply:
yeah, this is why it's kind of a tinfoil hat justification.